data_preprocessing.py

```
@@ -0,0 +1,45 @@
+df['label'] = df['label'].apply(lambda x: 0 if x == 'benign' else 1)
+
+# 3. 각 feature 계산
+df['hostname_length'] = df['url'].apply(lambda x: len(urlparse(x).netloc))
+df['count_dir'] = df['url'].apply(lambda x: urlparse(x).path.count('/'))
+df['count-www'] = df['url'].apply(lambda x: x.count('www'))
+df['url_length'] = df['url'].apply(lambda x: len(str(x)))
+
+def fd_length(url):
+    try:
+        return len(urlparse(url).path.split('/')[1])
+    except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)